src/new_index/schema.rs

```diff
@@ -512,6 +513,100 @@ impl ChainQuery {
         )
     }
 
+    pub fn summary(
+        &self,
+        scripthash: &[u8],
+        last_seen_txid: Option<&Txid>,
+        limit: usize,
+    ) -> Vec<TxHistorySummary> {
+        // scripthash lookup
+        self._summary(b'H', scripthash, last_seen_txid, limit)
+    }
+
+    fn _summary(
+        &self,
+        code: u8,
+        hash: &[u8],
+        last_seen_txid: Option<&Txid>,
+        limit: usize,
+    ) -> Vec<TxHistorySummary> {
+        let _timer_scan = self.start_timer("address_summary");
+        let rows = self
+            .history_iter_scan_reverse(code, hash)
+            .map(TxHistoryRow::from_row)
+            .map(|row| (row.get_txid(), row.key.txinfo))
+            .skip_while(|(txid, _)| {
+                // skip until we reach the last_seen_txid
+                last_seen_txid.map_or(false, |last_seen_txid| last_seen_txid != txid)
+            })
+            .skip_while(|(txid, _)| {
+                // skip the last_seen_txid itself
+                last_seen_txid.map_or(false, |last_seen_txid| last_seen_txid == txid)
+            })
+            .filter_map(|(txid, info)| {
+                self.tx_confirming_block(&txid)
+                    .map(|b| (txid, info, b.height, b.time))
+            });
+
+        // collate utxo funding/spending events by transaction
+        let mut map: IndexMap<Txid, TxHistorySummary> = IndexMap::new();
```

Comment:
Why use IndexMap? What's wrong with HashMap?

Reply:
IndexMap preserves insertion order, which means we get the transactions back out of the map in the same order their entries appeared in the history db.

We could use a plain HashMap and just re-sort the results at the end by e.g. height, but then we'd need some way to reproduce the correct ordering of transactions within the same block.

(although I'm not 100% sure if the history db ordering is correct in the first place - my first draft actually didn't use a map at all, and just accumulated matching funding/spending events directly in the iterator chain, but that didn't work because rows from different transactions were sometimes interleaved).

Reply:
Electrs currently can't tell the order of transactions or history entries in the same block.

Sorting via block number should be sufficient with the current guarantees. There should be mention of the fact that order within the same block is not stable.